examples/miscellaneous/plot_pipeline_display.py

```diff
@@ -0,0 +1,203 @@
+from sklearn.linear_model import LogisticRegression
+from sklearn.pipeline import make_pipeline
+import pandas as pd
+from sklearn import set_config
+
+numerical_columns_selector = selector(dtype_exclude=object)
+categorical_columns_selector = selector(dtype_include=object)
+
+data = pd.DataFrame()
+
+numerical_columns = numerical_columns_selector(data)
+categorical_columns = categorical_columns_selector(data)
+
+categorical_preprocessor = OneHotEncoder(handle_unknown="ignore")
+numerical_preprocessor = StandardScaler()
+
+preprocessor = ColumnTransformer(
+    [
+        ("one-hot-encoder", categorical_preprocessor, categorical_columns),
+        ("standard-scaler", numerical_preprocessor, numerical_columns),
+    ]
+)
+
+pipe = make_pipeline(preprocessor, LogisticRegression(max_iter=500))
+
+# %%
+# To visualize the diagram, change to `display='diagram'`
+set_config(display="diagram")
+pipe
+
+# %%
+# Displaying a Grid Search over a Pipeline
+###############################################################################
+# HTML Illustration of `GridSearchCV` over a `Pipeline` with `RandomForest`
+
+from sklearn.compose import make_column_selector as selector
+from sklearn.model_selection import GridSearchCV
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.impute import SimpleImputer
+from sklearn.preprocessing import OneHotEncoder, StandardScaler
+from sklearn.compose import ColumnTransformer
+from sklearn.pipeline import Pipeline
+from sklearn.pipeline import make_pipeline
+import numpy as np
+import pandas as pd
+from sklearn import set_config
+
+numerical_columns_selector = selector(dtype_exclude=object)
+categorical_columns_selector = selector(dtype_include=object)
+
+numeric_transformer = Pipeline(
+    steps=[
+        ("imputation_mean", SimpleImputer(missing_values=np.nan, strategy="mean")),
+        ("scaler", StandardScaler()),
+    ]
+)
+
+categorical_transformer = Pipeline(
+    steps=[
+        (
+            "imputation_constant",
+            SimpleImputer(fill_value="missing", strategy="constant"),
+        ),
+        ("onehot", OneHotEncoder(handle_unknown="ignore")),
+    ]
+)
+
+data = pd.DataFrame()
+
+numerical_columns = numerical_columns_selector(data)
+categorical_columns = categorical_columns_selector(data)
+
+preprocessor = ColumnTransformer(
+    transformers=[
+        ("num", numeric_transformer, numerical_columns),
+        ("cat", categorical_transformer, categorical_columns),
+    ]
+)
+
+pipe = Pipeline(
+    steps=[("preprocessor", preprocessor), ("classifier", RandomForestClassifier())]
+)
+
+param_grid = {
+    "classifier__n_estimators": [200, 500],
+    "classifier__max_features": ["auto", "sqrt", "log2"],
+    "classifier__max_depth": [4, 5, 6, 7, 8],
+    "classifier__criterion": ["gini", "entropy"],
+}
+
+grid_search = GridSearchCV(pipe, param_grid=param_grid, n_jobs=1)
+
+# %%
+# To visualize the diagram, change to `display='diagram'`
+set_config(display="diagram")
```

Comment:
Here as well, we probably don't need to call `set_config` every time, but perhaps the goal is to have self-contained cells?

Reply:
Right, it would be good to have the examples self-contained.